ports/psoc6/drivers/machine/psoc6_pwm.c

```diff
@@ -0,0 +1,29 @@
+// MTB includes
```

Comment:
I am still not fully convinced if this whole "driver" abstraction is really required. 
Not specifically for the PWM, but for all the modules.

Reply:
Sometimes I also feel that it's just adding an additional file & that could be avoided by including those in the machine module itself.  Since we started it as a practice from the initial module to have a separation between the cyhal apis and micropython just continued here.

Reply:
I agree. We can really get rid of the driver abstraction. The idea of having this was in cases where we are depending on the pdl layer and then API calls from mpy to cy was not 1:1. For example, if you see the psoc6_gpio, you realize that. But it does not have to be practice for all modules where it serves no benefit.